storage/client_test.go

```diff
@@ -337,16 +337,21 @@ func TestUpdateRangeAddressing(t *testing.T) {
 	for i, test := range testCases {
 		left := &proto.RangeDescriptor{RaftID: int64(i * 2), StartKey: test.leftStart, EndKey: test.leftEnd}
 		right := &proto.RangeDescriptor{RaftID: int64(i*2 + 1), StartKey: test.rightStart, EndKey: test.rightEnd}
+		var calls []*client.Call
 		if test.split {
-			if err := storage.SplitRangeAddressing(store.DB(), left, right); err != nil {
+			var err error
+			if calls, err = storage.SplitRangeAddressing(left, right); err != nil {
```

Comment:
I think the shared Run(calls) interface makes sense but how does introducing Batch resolve the Prepare/Flush issue? We still need to be able to prepare without flushing for the EndTransaction optimization (are you assuming the introduction of Commit() as mentioned above?)